### ISSUE TITLE
[no-unused-vars] positive in destructed array when another element is used 

### ISSUE THREAD

**Issue by chrisands** [typescript-eslint/typescript-eslint#3636]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": 2
  }
}
```

```TS
1: [
2:   ['', null],
3:   ['', true],
4:   ['', null],
5: ].filter(([_, second]) => second)
```

**Expected Result**

Expected on line 5 no error since I using `second` and if I remove `_` it will change element number.

**Actual Result**

error on line 5 that `_` is unused.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.24.0` |
| `@typescript-eslint/parser`        | `4.24.0` |
| `TypeScript`                       | `4.2.4` |
| `ESLint`                           | `7.26.0` |
| `node`                             | `10.21.0` |

**Comment by bradzacher** [comment]
Please read the docs for the rule.
There's a multiple options which can achieve what you want.
